### PR TITLE
Add PasswordField widget

### PR DESCRIPTION
**Pull request by gbaudic**
So, this is an old (8 years old) addition I made to my fork of guisan and for some reason, never contributed back to the main repo. Basically, it is just a text field but with hidden text, which is suitable for - you guessed it - passwords. 

I quickly updated it a few days ago to adjust to the upstream changes, but it may need some more improvements before merging. In particular, it may be good to make the masking character configurable through a setter.

### CODE REVIEW

File: include/guisan/widgets/passwordfield.hpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+         * Default constructor.
+         */
+        PasswordField();
+
+        /**
+         * Constructor. Initializes the passwordfield with a given string.
+         *
+         * @param text the initial text.
+         */
+        PasswordField(const std::string& text);
```
Comment (Jarod42) [inline include/guisan/widgets/passwordfield.hpp:90]:
```suggestion
        explicit PasswordField(const std::string& text);
```

File: src/widgets/passwordfield.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    PasswordField::PasswordField(const std::string& text) : TextField(text)
+    {
+
+    }
+
+
+    void PasswordField::draw(Graphics* graphics)
+    {
+        std::string encodedText(mText->getRow(0).size(), '*');
+        std::string realText(mText->getRow(0));
```
Comment (Jarod42) [inline src/widgets/passwordfield.cpp:85]:
```suggestion
        const std::string realText(mText->getRow(0));
        const std::string encodedText(realText.size(), '*');
```
Missing `const` and call `mText->getRow(0)` only once ;-)

File: src/widgets/passwordfield.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * For comments regarding functions please see the header file.
+ */
+
+#include "guisan/widgets/passwordfield.hpp"
+
+#include "guisan/font.hpp"
+#include "guisan/graphics.hpp"
+#include "guisan/key.hpp"
+#include "guisan/mouseinput.hpp"
+#include "guisan/text.hpp"
```
Comment (Jarod42) [inline src/widgets/passwordfield.cpp:67]:
```suggestion
#include "guisan/text.hpp"
```
Other headers should not be needed

File: include/guisan/widgets/passwordfield.hpp [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ */
+
+#ifndef GCN_PASSWORDFIELD_HPP
+#define GCN_PASSWORDFIELD_HPP
+
+#include "guisan/keylistener.hpp"
+#include "guisan/mouselistener.hpp"
+#include "guisan/platform.hpp"
+#include "guisan/widget.hpp"
+#include "guisan/widgets/textfield.hpp"
```
Comment (Jarod42) [inline include/guisan/widgets/passwordfield.hpp:64]:
```suggestion
#include "guisan/widgets/textfield.hpp"
```
Other headers unneeded.

`<string>` is discutable as include from "textfield.hpp", but not used in inherited interface.

**Review by midwan — APPROVED** [review]

File: include/guisan/widgets/passwordfield.hpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        
+        /**
+         * Set the masking character to hide the password
+         *
+         * @param mask the masking character
+         */
+        void setMaskingChar(const char mask);
+        
+    private:
+        char masking('*');
```
Comment (Jarod42) [inline include/guisan/widgets/passwordfield.hpp:100]: Not valid, 
```suggestion
        char masking = '*';
```

File: include/guisan/widgets/passwordfield.hpp [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        // Inherited from Widget
+
+        void draw(Graphics* graphics) override;
+        
+        /**
+         * Set the masking character to hide the password
+         *
+         * @param mask the masking character
+         */
+        void setMaskingChar(const char mask);
```
Comment (Jarod42) [inline include/guisan/widgets/passwordfield.hpp:97]: getter might be useful too.

**Review by Jarod42 — COMMENTED** [review]
Can you also add it in sample.

**Review by Jarod42 — APPROVED** [review]

**Pull request merged**